permissions_validators/src/public_blockchain/unregister.rs

```diff
@@ -17,35 +17,33 @@ declare_token!(
 #[derive(Debug, Copy, Clone, Serialize)]
 pub struct OnlyAssetsCreatedByThisAccount;
 
-impl_from_item_for_instruction_validator_box!(OnlyAssetsCreatedByThisAccount);
+impl IsAllowed for OnlyAssetsCreatedByThisAccount {
+    type Operation = Instruction;
 
-impl IsAllowed<Instruction> for OnlyAssetsCreatedByThisAccount {
     fn check(
         &self,
         authority: &AccountId,
         instruction: &Instruction,
         wsv: &WorldStateView,
-    ) -> Result<()> {
+    ) -> ValidatorVerdict {
         let unregister_box = if let Instruction::Unregister(unregister) = instruction {
             unregister
         } else {
-            return Ok(());
+            return ValidatorVerdict::Skip;
         };
-        let object_id = unregister_box
-            .object_id
-            .evaluate(wsv, &Context::new())
-            .map_err(|e| e.to_string())?;
-        let asset_definition_id: AssetDefinitionId = try_into_or_exit!(object_id);
+
+        let object_id = try_evaluate_or_deny!(unregister_box.object_id, wsv);
+        let asset_definition_id: AssetDefinitionId = ok_or_skip!(object_id.try_into());
         let registered_by_signer_account = wsv
             .asset_definition_entry(&asset_definition_id)
             .map(|asset_definition_entry| asset_definition_entry.registered_by() == authority)
             .unwrap_or(false);
         if !registered_by_signer_account {
-            return Err("Can't unregister assets registered by other accounts."
-                .to_owned()
-                .into());
+            return ValidatorVerdict::Deny(
```

Comment:
Consider adding a `fn deny(&self, reason: &str)` as an associated `fn`

Reply:
I think it's better to import `ValidatorVerdict::*` here to make code less clogged

Reply:
Yep. Just make sure it doesn't pollute the global scope.